lib/index.js

```diff
@@ -349,41 +353,45 @@ export default {
 				}
 				this.toolbar.update({ disableUI: false });
 				this.toolbar.hud.displayDefault();
+				this.checkForUpdates();
 			};
 
-			setTimeout(() => { // FIXME: Can we solve this without timeouts?
-				Appc.getInfo(() => {
-					const sdk = Appc.selectedSdk();
-					if (!sdk) {
-						const errorNotification = atom.notifications.addError('No Titanium SDK installed. Would you like to install the latest?', {
-							buttons: [
-								{
-									onDidClick: () => {
-										errorNotification.dismiss();
-										Appc.installSDK({
-											version: 'latest',
-											callback: (code) => {
-												if (code) {
-													atom.notifications.addWarning('Unable to install SDK');
-													return;
-												}
-												atom.notifications.addSuccess('Installed SDK');
-												Appc.getInfo(() => {
-													setupTargets();
-												});
-											}
-										});
-									},
-									text: 'Install latest SDK'
-								}
-							],
-							dismissable: true,
-						});
+			const { missing } = await Update.validateEnvironment();
+
+			if (missing.length) {
+				let message = 'You are missing the following required components for:';
+				for (let i = 0; i < missing.length; i++) {
+					const product = missing[i];
+					if (i < missing.length - 1) {
+						message = `${message} ${product.name},`;
 					} else {
-						setupTargets();
+						message = `${message} ${product.name}`;
 					}
+				}
+				message = `${message}. Without these components the extension will be unusable.`;
+
+				const errorNotification = atom.notifications.addError(message, {
+					buttons: [
+						{
+							onDidClick: async () => {
+								errorNotification.dismiss();
+								const updateInfo = [];
+								for (const product of missing) {
+									updateInfo.push(await product.getInstallInfo());
+								}
+								await Update.installUpdates(updateInfo, this.toolbar);
+								Appc.getInfo(() => {
+									setupTargets();
+								});
+							},
+							text: 'Install'
+						}
+					],
+					dismissable: true,
 				});
-			}, 1000);
+			} else {
+				setupTargets();
```

Comment:
Does this need to be performed after an `Appc.getInfo` call? I don't think we'll have the info otherwise?